win/WIN_ColourSlider.cpp

```diff
@@ -0,0 +1,50 @@
+#include "WIN_ColourSlider.h"
+
+using namespace win;
+
+ColourSlider::ColourSlider(SDL_Renderer* renderer, gfx::Rectangle rect, const char* name, gfx::Colour fillColour, gfx::Colour outlineColour, uint8_t* linkedVariable)
+	: Slider(renderer, rect, name, fillColour, outlineColour, 0, 0, 255)
+	, linkedVariable_(linkedVariable)
+{
+	setForegroundColour(fillColour);
+	setBackgroundColour(outlineColour);
+	positionFromValue(*linkedVariable_);
+}
+
+void ColourSlider::valueChangedByMovement()
+{
+	*linkedVariable_ = valueFromPosition();
+}
+
+void ColourSlider::valueChangedExternally()
```

Comment:
This is exactly the sort of thing that indicates an observer architecture is useful for us. We need to know that this function has to be called for the slider to operate correctly. Whereas a notification that propagates through the graph of components gets caught by the slider and it updates automatically.